src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,47 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  static alive = [];
+
+  constructor(name) {
+    this.name = name;
+    this.health = 100;
+    this.hidden = false;
+
+    Animal.alive.push(this);
+  }
+
+  reduceHealth(bitePower) {
+    if (!this.hidden) {
+      this.health -= bitePower;
+    }
+
+    if (this.health <= 0) {
+      this.removeAnimal();
+    }
+  }
+
+  removeAnimal() {
+    Animal.alive = Animal.alive.filter((animal) => animal.health > 0);
+  }
 }
 
 class Herbivore extends Animal {
-  // write your code here
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  bite(animal) {
+    const bitePower = 50;
+    const hidden = animal.hidden;
+    const herbivore = animal instanceof Herbivore;
```

Comment:
[CODE PERFORMANCE] - It's more efficient to check if the animal is an instance of Herbivore and not hidden within a single if statement to avoid unnecessary property access and improve performance.